src/components/BaseAutocomplete.vue

```diff
@@ -0,0 +1,102 @@
+<script setup lang="ts">
+import { computed, onBeforeUnmount, ref } from 'vue';
+
+type Option = { label: string; value: string; option?: Record<string, any> };
+
+const props = defineProps<{
+  options: Option[];
+  title: string;
+  modelValue?: string;
+}>();
+
+const optionsEl = ref<Node | undefined>(undefined);
+const inputEl = ref<Node | undefined>(undefined);
+const displayDropdown = ref(false);
+
+const getInputLabel = (value?: string) => {
+  const option = props.options.find(option => option.value === value);
+  return option ? option.label : '';
+};
+
+const inputValue = ref(getInputLabel(props.modelValue));
+
+const displayedOptions = computed(() => {
+  if (!props.options) return [];
+  return props.options.filter(option =>
+    option.label.toLowerCase().includes(inputValue.value.toLowerCase())
+  );
+});
```

Comment:
Might not be really practicable in this case but we have this already:

https://github.com/snapshot-labs/snapshot/blob/1ca15e8ea71a2b1eb20afc0616f47f754c4ec3af/src/composables/useNetworksFilter.ts#L16-L25

So we have two functions that filter networks by user input now. But maybe anything other than that would require more work than reasonable. I guess the actual issue this arises from lies a lot deeper... like... refactoring hell deep! :D So just don't mind maybe.

Reply:
Yeah, I had seen that, but since I filter inside autocomplete component, it didn't make sense for me to filter outside of it. 
Maybe I can expose a custom filter method on the autocomplete component where we can pass this method.